### PR TITLE
Docker tags

### PR DESCRIPTION
**Pull request by SwampDragons**
Change field name of docker tag to "tags" instead of "tag" since it's a list. Keep "tag" for backwards compatability.

### CODE REVIEW

File: post-processor/docker-tag/post-processor.go [hunk 1]
```diff
@@ -47,11 +49,23 @@ func (p *PostProcessor) Configure(raws ...interface{}) error {
...
+
 	return nil
 
 }
 
 func (p *PostProcessor) PostProcess(ctx context.Context, ui packer.Ui, artifact packer.Artifact) (packer.Artifact, bool, bool, error) {
+	if len(p.config.Tag) > 0 {
+		ui.Say("Deprecation warning: \"tag\" option has been replaced with " +
+			"\"tags\". In future versions of Packer, this configuration may " +
+			"not work. Please call Packer Fix against your template ")
```
Comment (azr) [inline post-processor/docker-tag/post-processor.go:66]: Nice !

File: website/pages/docs/post-processors/docker-tag.mdx [hunk 2]
```diff
@@ -1,4 +1,4 @@
----
+]---
```
Comment (azr) [inline website/pages/docs/post-processors/docker-tag.mdx:1]: Is this on purpose ?

Comment (azr) [reply website/pages/docs/post-processors/docker-tag.mdx:1]:
```suggestion
---
```

**Comment by SwampDragons** [comment]
Good point Moss -- this isn't backwards incompatible but it probably should get a fixer now instead of whenever we remove the "tag" option for real.

File: fix/fixer_pp_docker_tag_tags.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
+package fix
+
+import (
+	"strings"
+
+	"github.com/mitchellh/mapstructure"
+)
+
+// FixerDockerTagtoTags renames any Filename to Output
```
Comment (nywilken) [inline fix/fixer_pp_docker_tag_tags.go:9]:
```suggestion
// FixerDockerTagtoTags renames tag to tags
```

File: fix/fixer_pp_docker_tag_tags.go [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+			// Gather all "tag" into the []string
+			switch t := tagRaw.(type) {
+			case []interface{}:
+				for _, tag := range t {
+					allTags = append(allTags, tag.(string))
+				}
+			case []string:
+				allTags = append(allTags, t...)
+			case string:
+				tList := strings.Split(t, ",")
```
Comment (nywilken) [inline fix/fixer_pp_docker_tag_tags.go:57]: Should we strip white space here? Thinking in the case of `"tag": "one, two"`

File: fix/fixer_pp_docker_tag_tags.go [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+		delete(pp, "tag")
+		pp["tags"] = finalTags
+	}
+
+	input["post-processors"] = tpl.PostProcessors
+	return input, nil
+}
+
+func (FixerDockerTagtoTags) Synopsis() string {
+	return `Updates "manifest" post-processor so any "filename" field is renamed to "output".`
```
Comment (nywilken) [inline fix/fixer_pp_docker_tag_tags.go:103]: This needs to be updated to match this fixer.

Comment (SwampDragons) [reply fix/fixer_pp_docker_tag_tags.go:103]: oops thanks

File: fix/fixer_pp_docker_tag_tags.go (hunk 4 above)
Comment (SwampDragons) [reply fix/fixer_pp_docker_tag_tags.go:57]: good point

**Pull request merged**